dxcint/dxcint/Dependency.py

```diff
@@ -0,0 +1,200 @@
+import os
+import re
+import json
+import tarfile
+import subprocess as sp
+
+from typing import Dict
+from pathlib import Path
+
+from dxcint.Context import Context
+
+
+class DependencyError(Exception):
+    """
+    Class to handle Dependency exceptions
+    """
+
+
+class DependencyFactory(object):
+    def __init__(self, config_file: str, context: Context):
+        self._type_switch = {
+            "binary": BinaryDependency,
+            "tarball": TarballDependency,
+            "package_manager": PackageDependency,
+            "default": Dependency
+        }
+        self._config_file = config_file
+        self._context = context
+
+    def make(self):
+        detected_type = self._detect_type()
+        return self._type_switch[detected_type](self._config_file, self._context)
+
+    def _detect_type(self) -> str:
+        """
+        This logic defines the precedence of detection. First precedence: If config has `package_manager` field, then
+        it's a package dependency. Later this will be reconsidered to always deliver dependencies as packages
+
+        Returns: str. Type of package.
+        """
+        with open(self._config_file, "r") as config_handle:
+            config = json.load(config_handle)
+        if config.get("package_manager", None):
+            return "package_manager"
+
+        src_link = config.get("source_link", None)
+        if src_link:
+            if re.search(r"tar\..z2?$", src_link):
```

Comment:
Is it possible that the tarball format will be tgz?

Reply:
sure, will add that if we add a dependency which is delivered as tgz